src/main/java/seedu/address/logic/commands/FindRoleCommand.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.person.RoleContainsKeywordsPredicate;
+
+
+/**
+ * Finds and lists all persons in address book whose role matches the argument roles.
+ */
+public class FindRoleCommand extends Command {
+
+    public static final String COMMAND_WORD = "findrole";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose role matches the specified "
+            + "role and displays them as a list with index numbers.\n"
+            + "A role must be either CLIENT, SUPPLIER or EMPLOYEE. \n"
+            + "Parameters: ROLE [ROLES]...\n"
+            + "Example: " + COMMAND_WORD + " CLIENT SUPPLIER";
```

Comment:
Seems like the ROLE parameter is case sensitive, and only accepts `client`, `supplier` or `employee` in all lower case. Checked by `FindRoleCommandParser` using `Person#isValidRole()`